kvs: Support ability to list namespaces

chu11 opened the pull request:
Support a simple iterator to allow users to list the KVS namespaces that exist, along with the owner and current flags.  The ```flux kvs namespace-list``` option would list something like this:

```
NAMESPACE             OWNER      FLAGS
primary                8556 0x00000000
```

chu11 commented:
hmmm,  a stalled build, last line was

```
PASS: ../t/t9990-python-tests.t 44 test.request.TestRequestMethods.test_null_payload
```

this one appears to be new.  Just jotting it down, will restart build.

garlick commented:
I'd probably flatten the new service method name, since the extra period implies a module hierarchy (see intro in [RFC 5](https://github.com/flux-framework/rfc/blob/master/spec_5.adoc)), e.g. use `kvs.listns` or similar.

It might be a good idea to have a name column wide enough for a uuid in [RFC 4122](https://tools.ietf.org/html/rfc4122) ascii form, e.g.
```
f81d4fae-7dec-11d0-a765-00a0c91e6bf6
```
since it's handy to create a unique KVS namespace that way, if no other source of uniqueness is available.

chu11 commented:
Just re-pushed with spacing enough to handle uuid width.

Noticed coverage didn't hit 78%, so I reworked some code in the kvs module.  There is no need to handle the "zero namespaces" path, since ```json_pack()``` will do the smart thing with an empty array.  

Changes are tiny and went ahead and squashed it all.

> I'd probably flatten the new service method name

I see your point.  How about we clean that up within the context of #1302 instead.  Because I should probably correct that in a lot of places (```kvs.stats.get```, ```kvs.namespace.create```, ```kvs.namespace.remove```, ```kvs.setroot.*```, etc.).  I'm thinking underscores/dashes or something instead of periods.

garlick commented:
Sure that's fine by me.

garlick commented:
Just checked this out and played with it a bit - nice work.  Are you still trying to hit 78% or do you think it's ready to go?

chu11 commented:
I think its good to go.  If I wrote a whole new C file test I think I could only eek out one more line of coverage, so doubt I'm going to get to 78%.
